### PR TITLE
Update CI to match module template

### PR DESCRIPTION
**Pull request by Mrtenz**
This updates CI to be closer to the current module template. There are some minor differences, since this repo is used as template.

### CODE REVIEW

File: scripts/cleanup.sh [hunk 1]
```diff
@@ -6,5 +6,6 @@ set -o pipefail
 rm .github/CODEOWNERS
 rm .github/workflows/create-release-pr.yml
 rm .github/workflows/publish-release.yml
+rm .github/workflows/publish-site.yml
```
Comment (Mrtenz) [inline scripts/cleanup.sh:9]: I'm not sure if this still makes sense. It will cause the `main.yml` workflow to be broken, probably.

File: .nvmrc [hunk 2]
```diff
@@ -1 +1 @@
-v18.6
+lts/*
```
Comment (FrederikBolding) [inline .nvmrc:1]: Maybe we want to keep more control of this because it is consumer facing?

File: .github/workflows/publish-site.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
+name: Publish site to GitHub Pages
```
Comment (FrederikBolding) [inline .github/workflows/publish-site.yml:1]: I think we can remove all the publishing logic tbh?

File: .github/workflows/publish-release.yml [hunk 4]
```diff
@@ -1,48 +1,88 @@
 name: Publish Release
```
Comment (FrederikBolding) [inline .github/workflows/publish-release.yml:1]: I think we can remove all the publishing logic tbh?

File: .github/workflows/create-release-pr.yml [hunk 5]
```diff
@@ -8,7 +8,7 @@ on:
         default: 'main'
```
Comment (FrederikBolding) [inline .github/workflows/create-release-pr.yml:8]: Perhaps this can be removed too?

**Review by FrederikBolding — APPROVED** [review]

**Pull request merged**